### DISCUSSION TITLE
Can we access native graphics API

### DISCUSSION THREAD

**Discussion by hamyyy** [DiligentGraphics/DiligentEngine#184] [Q&A] [answered]
I'm trying to integrate [Optick ](https://github.com/bombomby/optick/wiki/Optick-API#optick_gpu_init_d3d12device-cmd_queues-num_cmd_queus)(CPU/GPU profiler) into my project. 
In order to get the GPU profiling to work, I need a reference to D3D/Vulkan Device object as well as the command queue.
How can I get the native API  methods/variables from Diligent?

**Comment by TheMostDiligent** [comment, marked as answer]
For every object you can query API-specific interface.
https://github.com/DiligentGraphics/DiligentCore/blob/master/Graphics/GraphicsEngineD3D11/interface/RenderDeviceD3D11.h#L57
https://github.com/DiligentGraphics/DiligentCore/blob/master/Graphics/GraphicsEngineVulkan/interface/RenderDeviceVk.h#L57
https://github.com/DiligentGraphics/DiligentCore/blob/master/Graphics/GraphicsEngine/interface/DeviceContext.h#L3122
https://github.com/DiligentGraphics/DiligentCore/blob/master/Graphics/GraphicsEngineVulkan/interface/CommandQueueVk.h#L80
